### DISCUSSION TITLE
Unlisted posts - do they do what we expect them to do and what we want them to do?

### DISCUSSION THREAD

**Discussion by strypey** [mastodon/mastodon#21652] [Q&A]
I've become really confused about what an unlisted post does. Both what it does Mastodon>Mastodon, and whether that's different from what it does when received by any of the other ActivityPub implementations (but maybe that's a topic for somewhere like SocialHub). For now, I'll talk about it in general terms.

At the start of this month, I thought I could use unlisted posts to have loads of side conversations, without flooding the people following my account. I also though that hashtags still worked in unlisted posts. Which I liked because I thought I could spare the people following me long strings of posts on a niche topic, while still making them available to anyone doing hashtag searches for the kind of stuff in them.

So for example, I could have a long discussion with one or two people about the airspeed velocity of an unladen swallow, using unlisted posts, and the only people who would see them were:

* the people @mentioned (assuming their mentions aren't a firehose ;)

* people who clicked on the original public post and followed the thread down.

* people who found posts in the thread by searching on (or following) hashtags like #swallow, or the names of countries swallows migrate between.

I'm pretty clear on the fact that unlisted posts don't appear in the Local or Federated timelines. But;

* do they appear in the Home timelines of people following my account?

* do they appear in hashtag searches for tags they contain?

Can anyone point me to an authoritative source that confirms what the current behaviour of unlisted posts is? If it's not what I thought it was? Why is that? Are my expectations unreasonable from either a technical or UX perspective?

Any clarification or thoughts on this would be most appreciated.

**Comment by trwnh** [comment]
The ONLY thing "Unlisted" does is:

- Your post will not show up in public timelines (Local, Federated, Hashtag)

That's it. The only reason "Unlisted" exists is because the public timelines exist, and people needed a way to opt out.

HOWEVER: There are additional filtering rules on Home and Public timelines, unrelated to the above:

- Replies will be hidden unless they are self-replies within the first chain (i.e. no one else has replied up-thread)
  - Replies to people you follow will be visible in your Home timeline

At some point around this latest Twitter migration wave, it became clear that a lot of people (especially new people) believed any number of untrue additional properties about what "Unlisted" does to your posts.

This isn't the "current" behavior, either -- they've *always* been like this.

Some myths I've seen being propagated:

- Myth: Making your reply unlisted will cause it to not show up in the public timeline.
  - Fact: Your reply wouldn't show up anyway unless it was a chain of self replies
- Myth: Making your reply unlisted will cause it to not show up in your followers' home timelines.
  - Fact: Your reply will show up regardless, unless you are replying to someone that they do not follow

It should be noted that other software such as Pleroma/Akkoma/Misskey does not filter out replies -- users of those software will generally see all replies in their home timelines, and all public posts in the public timelines with no extra conditions. (This is a configurable setting, though.)

To address specific questions you had:

> the people @mentioned (assuming their mentions aren't a firehose ;)

Yes

> people who clicked on the original public post and followed the thread down.

Yes

> people who found posts in the thread by searching on (or following) hashtags like #swallow, or the names of countries swallows migrate between.

No -- hashtag timelines are considered public timelines, so unlisted posts will not show up there, even if they contain hashtags. This is to allow talking *about* a hashtag without being included *in* that hashtag.

> do they appear in the Home timelines of people following my account?

Yes

> do they appear in hashtag searches for tags they contain?

No

---

I can't speak to *why* people are developing these specific expectations, but I do agree that it is not as clear as it could be. The language in the UI states "Visible to all, but opted out of discovery features", which to me seems clear, but "discovery features" is a bit vague and ill-defined. The only other alternative to "Unlisted" I can think of right now would be "Undiscoverable", but that could be more confusing and make people think it was invisible in general. Perhaps you could just call it "Not Public", but that also has issues based on what people think "Public" means. The most precise way to describe it would be "Not in public timelines", I suppose?

**Reply by 2xaronl** [reply]
I have a Use-Case related to this topic and I'd like to know if there's a way to get it to work:

I want to "followers-only" post to my followers and use hashtags so they can easily search for different subjects I'm posting about.  Currently, it seems that the hashtag search only works if I post publicly.  Even if a follower follows the hashtag I used in a "followers-only'" post they cannot find that post by hashtag search.

Reading the above, it seems like this won't work by-design.  Is there a way to enable this as an optional feature?

I enabled Elasticsearch full-text search but that doesn't seem to address this issue either.
